build.gradle

```diff
@@ -42,6 +43,29 @@ ext {
   logger.info("Is cloudbees? $isCloudbees")
 }
 
+// Java 6 build setup
+def java6Error = 'Set the poperty \'java6Home\' in your $HOME/.gradle/gradle.properties pointing to a Java 6 installation'
+assert (JavaVersion.current().isJava8Compatible()) : 'The Ehcache 3 build requires Java 8 to run and a configured Java 6 installation\n' + java6Error
+assert hasProperty('java6Home') : 'The Ehcache 3 build requires a configured Java 6 installation\n' + java6Error
+def java6ExecutablesPath = new File(java6Home)
+def testExectuablePath = java6ExecutablesPath
```

Comment:
Funny - will rename them all - no longer point to `bin` but effectively to java home 😉
I pushed the `bin` concern down to the build utilities class.